content/cpp/concepts/pointers/pointers.md

```diff
@@ -11,32 +11,81 @@ CatalogContent:
   - 'paths/computer-science'
 ---
 
-A **pointer** is a special type of variable that stores the address of another variable in it.
+A **pointer** is a special type of variable that stores the memory address of another variable.
 
 ## Declaration and Initialization of Pointer variable
 
-A pointer variable points to the same data type and is created with the `*` operator. Then the address of another variable is stored in it.
+A pointer variable is declared using the indirection operator (`*`). This is a special unary operator that signals the program to store the memory address of another variable.
+
+To get the memory address of a variable to store in the pointer variable, the address operator (`&`) is used. This signals the program to return the memory address of a variable. Memory addresses are usually prefixed with `0x` this just indicates that the following digits are in hexadecimal notation.
 
 ```cpp
 std::string game = "Cricket";
 
-// A pointer with the name ptr, that stores the address of game
-std::string* ptr = &game;
-
 std::cout << game << "\n";   // Output: Cricket
 std::cout << &game << "\n";  // Output of game memory address: 0x6dfed4
+
+// A pointer with the name ptr, that stores the address of game
+std::string* ptr = &game;
 std::cout << ptr << "\n";    // Output of pointer to game: 0x6dfed4
 ```
 
 ## Pointer Dereferencing
 
 Dereferencing a pointer means getting the value that a pointer variable is pointing at.
 
-The dereference operator `*` is used to get the value stored in the memory location pointed by the pointer:
+There aren't many of useful operations that can be performed using just the memory address that pointers store. Instead, there needs to be a way to get the value stored in the memory address so it can be manipulated directly.
+Dereferencing a pointer helps to do this. A pointer is derefrenced with the indirection operator (`*`) as well.
```

Comment:
```suggestion
This can be done by dereferencing a pointer. The indirection operator (`*`) used for declaring a pointer can be used to dereference a pointer as well.
```